addon/components/toolbar/mark.ts

```diff
@@ -8,25 +9,38 @@ type Args = {
   mark: string;
 };
 export default class MarkComponent extends Component<Args> {
+  constructor(owner: unknown, args: Args) {
+    if (!args.controller?.schema.marks[args.mark]) {
```

Comment:
```suggestion
    if (!args.controller.schema.marks[args.mark]) {
```

Reply:
This suggestion would just cause it to fail to start as `args.controller` will be undefined. I'll tweak this to not log on startup. I don't really want to change the way this is used just for this convenience log to work correctly.

Reply:
True, but then we should also adjust the argument types to mark the `controller` argument as optional.

Reply:
Hmm, this is tricky. In this kind of context, TS assumes that your code is obeying the types, which in a pure TS codebase should (but won't always) be true. At the very least, there should be an error when you misuse something. With a mixture of TS and JS or TS and handlebars, this is no longer true.

I don't think we should change the arguments as while it no longer should crash if `controller` is not supplied, the class is useless if it isn't. I can't think of a good way to, within this class body, treat `Args` as partial, but keep the members of `Args` as required. Just marking `get controller()` as returning `SayController | undefined` for example does not seem to work.

Reply:
why would controller be undefined?
that would mean the parent component is passing in an incorrect value, so crashing is totally fine.
Sure, you'd like a build-time warning which we loose because we pass through unchecked templates (which we'll get when we can enable glint), but that doesn't change how we should design the component imo. 
Either you support the flexibility, and type it as optional, and nullcheck yourself, or you don't, and require the calling component to make sure its not passing in undefined.